static/third/bootstrap-typeahead/typeahead.js

```diff
@@ -141,7 +156,12 @@
         // converts the input text to html elements.
         this.$element.html('');
       } else {
-        this.$element.val(this.updater(val, e)).trigger("change");
+        const after_text = this.updater(val, e);
+        const [from, to_before, to_after] = get_string_diff(this.$element.val(), after_text);
+        const replacement = after_text.substring(from, to_after);
+        // select / highlight the minimal text to be replaced
+        this.$element[0].setSelectionRange(from, to_before);
+        insert(this.$element[0], replacement);
```

Comment:
What happens when you're, say, accepting a completion that is already directly present in the message, and thus the insertion might be a noop? I'm concerned by this bit of code from reading `node_modules/text-field-edit/index.js` that it'll do some sort of deletion operation?

```
function safeTextInsert(text) {                                                                          
    if (text === '') {                                                                                   
        return document.execCommand('delete');                                                           
    }                                                                                                    
    return document.execCommand('insertText', false, text);                                              
}                                                                                                        
```

Reply:
Thanks for pointing out this edge case! 

The delete in the code quoted above is for deleting any selected text before inserting new text in it's place.

Though I had already accounted for cases where the before and after strings are identical (in other words where insertion is noop), I realised a similar corner case comes up for slash commands, where even if you have the entire command typed out and then accept the completion, it adds a space right after. This indeed was buggy, but simply ensuring that diff_end >= diff_start has fixed this now. Have also added a test for the same.